Standards/scs-0217-v1-baseline-cluster-security.md

```diff
@@ -0,0 +1,145 @@
+| TCP      | 10251/10259 | kube-scheduler          | 10251 could be insecure before 1.13, after that only the secure port 10259 is available |
+| TCP      | 10252/10257 | kube-controller-manager | 10252 could be insecure before 1.13, after that only the secure port 10257 is available |
+| TCP      | 10255       | Read-Only Kubelet API   | HTTP port for GET requests                                                              |
+| TCP      | 30000-32767 | NodePort Services       | Service endpoints, could be HTTP                                                        |
+
+#### Authentication and Authorization
+
+All API clients should authenticate and authorize in order to be able to access an API or even
+specific functions of this API. This is the case for users as well as internal components.
+
+Most internal clients (like proxies or nodes) are typically authenticated via service accounts or
+x509 certificates, which will normally be created automatically during the setup of a cluster.
+External users can authenticate via an access pattern of choice, which is typically decided by
+the cluster provider.
+
+Authorization is (normally) done by the Role-Based Access Control (RBAC), which matches a request
+by a user with a set of permissions, also called a role. Kubernetes deploys some roles out-of-the-box;
+additional roles need to be carefully checked, since some permissions for specific resources allow
+modification of other resources.
+
+This whole process is especially important for the Kubelet, which allows anonymous requests in its
+default configuration. This is obviously a security risk, since everybody with access to its endpoint
+could manipulate resources that are managed with the Kubelet.
+
+To disable anonymous requests, the Kubelet should be started with `--anonymous-auth=false`.
+Authentication can be provided either through x509 client certificates or API bearer tokens.
+How to set up both approaches can be found in the [Kubelet Authentication and Authorization](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-authn-authz/).
+
+Kubelet authorization is set to `AlwaysAllow` as a default mode. This can be quite problematic,
+since all authenticated users can do all actions. To mitigate this, it is possible to delegate
+authorization to the API server by:
+
+- enabling the `authorization.k8s.io/v1beta1` API group
+- starting the Kubelet with the `--authorization-mode=Webhook` and the `--kubeconfig` flags
+
+After that, the Kubelet calls the `SubjectAccessReview` API in order to determine the authorization of a request.
+
+## Decision
+
+This standard tries to increase security for a Kubernetes cluster in order to provide a
+solid baseline setup with regard to security. For this to work, multiple measures need to be undertaken.
+
+A self-controlled CA SHOULD be used in order to be in control of the TLS certificates, which
+enables the operator to provide and revoke certificates according to the requirements.
```

Comment:
according to which requirements? Requirements of this or another standard? Requirements of the cloud operator?

Please describe the requirements clearly. Also, if there are any clear requirements, please explain why these need a custom internal CA and not a standard browser CA.

Reply:
There are neither operator requirements nor requirements about CAs in the linked issue, other than that an unspecified "external CA" should be used.

I guess we can move on and don't change this, as there don't actually seem to be any written down requirements for CAs anywhere here which we could explicitly mention, especially with regards to security.